### PR TITLE
Replace never() calls in BehaviorSubjectTest with verifyNoMoreInteractions

### PR DESCRIPTION
**Pull request by vanniktech**
Advantage is that we verify that there's no interaction at all so also something that we might not expect is not going to happen.

### CODE REVIEW

File: src/test/java/rx/subjects/BehaviorSubjectTest.java [hunk 1]
```diff
@@ -43,7 +44,7 @@
 
 public class BehaviorSubjectTest {
 
-    private final Throwable testException = new Throwable();
+    private Throwable testException = new Throwable();
```
Comment (akarnokd) [inline src/test/java/rx/subjects/BehaviorSubjectTest.java]: Why did this lose `final`?

Comment (vanniktech) [inline src/test/java/rx/subjects/BehaviorSubjectTest.java]: @akarnokd was a mistake. I added the final back

**Comment by akarnokd** [comment]
:+1:

**Comment by stevegury** [comment]
:+1:

**Pull request merged**